src/utils/daos_hdlr.c

```diff
@@ -2887,6 +3086,595 @@ fs_copy_hdlr(struct cmd_args_s *ap)
+					eprs, &recx_anchor,
+					true, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to list recx: %d\n", rc);
+			D_GOTO(out, rc);
+		}
+
+		/* if no recx is returned for this dkey/akey move on */
+		if (number == 0)
+			continue;
+
+		D_ALLOC(buf, number);
+		if (buf == NULL) {
+			D_GOTO(out, rc = -DER_NOMEM);
+		}
+
+		d_iov_t	iov[number];
+
+		/* set iod values */
+		(*iod).iod_type  = DAOS_IOD_ARRAY;
+		(*iod).iod_nr    = number;
+		(*iod).iod_recxs = recxs;
+		(*iod).iod_size  = size;
+
+		/* set sgl values */
+		sgl.sg_nr_out = 0;
+		sgl.sg_iovs   = iov;
+		sgl.sg_nr     = number;
+
+		for (i = 0; i < number; i++) {
+			buf_len = recxs[i].rx_nr * size;
+			D_ALLOC(buf[i], buf_len);
+			if (buf[i] == NULL) {
+				/* free any other buf[i] that
+				 * was previously allocated
+				 * on error
+				 */
+				for (k = 0; k < i; k++) {
+					D_FREE(buf[k]);
+				}
+				D_FREE(buf);
+				D_GOTO(out, rc = -DER_NOMEM);
+			}
+			d_iov_set(&iov[i], buf[i], buf_len);
+		}
+
+		/* fetch recx values from source */
+		rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, dkey,
+				    1, iod, &sgl, NULL, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to fetch source "
+				"recx: %d\n", rc);
+			D_GOTO(err_out, rc);
+		}
+
+		/* update fetched recx values and place in
+		 * destination object
+		 */
+		rc = daos_obj_update(*dst_oh, DAOS_TX_NONE, 0, dkey,
+				     1, iod, &sgl, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to update destination "
+				"recx: %d\n", rc);
+			D_GOTO(err_out, rc);
+		}
+		for (i = 0; i < number; i++) {
+			D_FREE(buf[i]);
+		}
+		D_FREE(buf);
+	}
+	D_GOTO(out, rc);
+err_out:
+	for (i = 0; i < number; i++) {
+		D_FREE(buf[i]);
+	}
+	D_FREE(buf);
+out:
+	return rc;
+}
+
+static int
+cont_clone_list_akeys(daos_handle_t *src_oh,
+		      daos_handle_t *dst_oh,
+		     daos_key_t diov)
+{
+	int		rc = 0;
+	int		j = 0;
+	char		*ptr;
+	daos_anchor_t	akey_anchor = {0};
+	daos_key_desc_t	akey_kds[ENUM_DESC_NR] = {0};
+	uint32_t	akey_number = ENUM_DESC_NR;
+	char		*akey = NULL;
+	d_sg_list_t	sgl;
+	d_iov_t		iov;
+	daos_key_t	aiov;
+	daos_iod_t	iod;
+	char		*small_key = NULL;
+	char		*large_key = NULL;
+	char		*key_buf = NULL;
+	daos_size_t	key_buf_len = 0;
+
+	D_ALLOC(small_key, ENUM_DESC_BUF);
+	if (small_key == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+	D_ALLOC(large_key, ENUM_LARGE_KEY_BUF);
+	if (large_key == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	/* loop to enumerate akeys */
+	while (!daos_anchor_is_eof(&akey_anchor)) {
+		memset(akey_kds, 0, sizeof(akey_kds));
+		memset(small_key, 0, ENUM_DESC_BUF);
+		memset(large_key, 0, ENUM_LARGE_KEY_BUF);
+		akey_number = ENUM_DESC_NR;
+
+		sgl.sg_nr     = 1;
+		sgl.sg_nr_out = 0;
+		sgl.sg_iovs   = &iov;
+
+		key_buf = small_key;
+		key_buf_len = ENUM_DESC_BUF;
+		d_iov_set(&iov, key_buf, key_buf_len);
+
+		/* get akeys */
+		rc = daos_obj_list_akey(*src_oh, DAOS_TX_NONE, &diov,
+					&akey_number, akey_kds,
+					&sgl, &akey_anchor, NULL);
+		if (rc == -DER_KEY2BIG) {
```

Comment:
future improvement:
this should be a continued try with larger buffer if KEY2BIG scenario, because we don't impose a limitation on the key buffer size (for akeys and dkeys).
but this is good enough for a first cut.